### PR TITLE
[sprites] New Indicator For Turrets

### PR DESCRIPTION
**Pull request by Voz-Duh**
I have tried to accommodate all styles of game and I hope I succeeded.
new-turrets

### CODE REVIEW

**Comment by GlennFolker** [comment]
Why not make fuse's blue entirely as the indicator?

**Comment by Voz-Duh** [comment]
> Why not make fuse's blue entirely as the indicator?

It looks bad, I already tried

File: core/src/mindustry/world/blocks/defense/turrets/Turret.java [hunk 1]
```diff
@@ -75,6 +75,15 @@
...
 
     public Cons<TurretBuild> drawer = tile -> Draw.rect(region, tile.x + tr2.x, tile.y + tr2.y, tile.rotation - 90);
+    public Cons<TurretBuild> teamDrawer = tile -> {
+        if(tile.team.id == 1 || tile.team.id == 2){
+            Draw.rect(teamRegions[tile.team.id], tile.x + tr2.x, tile.y + tr2.y, tile.rotation - 90);
+        }else{
+            Draw.color(tile.team.color);
+            Draw.rect(teamRegion, tile.x + tr2.x, tile.y + tr2.y, tile.rotation - 90);
+        }
+        Draw.color();
```
Comment (summetdev) [inline core/src/mindustry/world/blocks/defense/turrets/Turret.java:85]:
```suggestion
        }else{
            Draw.color(tile.team.color);
            Draw.rect(teamRegion, tile.x + tr2.x, tile.y + tr2.y, tile.rotation - 90);
            Draw.color();
        }
```

**Comment by sk7725** [comment]
Make the player's team's turrets appear as normal and color the rest.

**Comment by the-red-soul** [comment]
to rephrase waht sk said, potentially leave the Sharded team the default colours instead of having them changed up.

**Comment by sample-text-here** [comment]
It looks good, but here are a few things I would change
- add colors to the duo, hail, and scorch (color the white parts?)
- make the salvo's and the wave's blue the team color instead (right now they look messy)
- make the white part of the scatter have color instead
- make it easier to see the meltdown and spectre's team colors
- (maybe) color the cyclone's yellow and ripple's orange parts?

**Comment by SummetDev** [comment]
@Voz-Duh why closed?

**Pull request closed without merging**